### PR TITLE
chore: Improve share dialog behavior

### LINKED ISSUE #14601: [Bug]: Share modal in view mode opens up multiple modals and appears glitchy
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Share modal in view mode opens up multiple modals and appears glitchy. This does not happen all the time but this behaviour can be observed by repeating the action a couple of times 

### Steps To Reproduce

1. Go to view mode of any app and click on the share button to see a glithcy behaviour as seen in the video

https://www.loom.com/share/c66bb2e72b1440a3b692867d3b6aebe5glitchy

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by nzidol**
## Description
Screen updates during 3 separate api calls looks glitchy. Use async/await to run all 3 before a redraw.

Fixes #14601

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Cypress test

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Rhitottam** [comment]
/perf-test ref=rhito/feat-create-run-meta-before-test-starts-compare-lt-st-for-prev-runs

**Comment by ankitakinger** [comment]
@nzidol Thank you for your contribution. I was checking this change on my local machine and in spite of this change, the issue doesn't seem to be resolved as I can still see this happening. Could you maybe attach a before and after recording for this change, to show how this improves the share dialog behavior?

Also, your PR has conflicts that need to be resolved as well.

**Comment by nzidol** [comment]
Hi,

as explained here [https://github.com/appsmithorg/appsmith/issues/14601](https://github.com/appsmithorg/appsmith/issues/14601 ) this resolves half the problem. It reduces the dialog redraws from 5 to 3;  to see this you need to capture the process with cypress and step through the results. To resolve the second issue requires a coding standard decision on where to place React hooks when embedding a component inside another.

Not sure why there was a merge conflict, it just highlighted the required changes....

File: app/client/src/ce/pages/workspace/WorkspaceInviteUsersForm.tsx [hunk 1]
```diff
@@ -355,9 +355,14 @@ function WorkspaceInviteUsersForm(props: any) {
...
   useEffect(() => {
-    fetchUser(props.workspaceId);
-    fetchAllRoles(props.workspaceId);
-    fetchCurrentWorkspace(props.workspaceId);
+    const fetchAll3 = async () => {
+      await fetchUser(props.workspaceId);
+      await fetchAllRoles(props.workspaceId);
+      await fetchCurrentWorkspace(props.workspaceId);
+    };
+    if (!isAclFlow) {
```
Comment (ankitakinger) [inline app/client/src/ce/pages/workspace/WorkspaceInviteUsersForm.tsx:363]: Hey @nzidol, could you re-check on this, the props isAclFlow has been removed. This should be causing an error on the client side locally.

Comment (nzidol) [reply app/client/src/ce/pages/workspace/WorkspaceInviteUsersForm.tsx:363]: Looks like there has been significant change in the module, so I'll withdraw the PR. Just note that the React hooks are the issue here, the 3api's responding at different times will cause dialog redraws.

**Comment by nzidol** [comment]
Needs further review given large amount of change in module.

**Pull request closed without merging**